### ISSUE TITLE
Redesign the header to be consistent across all pages

### ISSUE THREAD

**Issue by sainak** [coronasafe/care_fe#7316]
Currently the header is very inconsistent and space consuming on the screen so, to fix that make the header a reusable component

current headers across pages
<img width="1161" alt="image" src="https://github.com/coronasafe/care_fe/assets/46787056/2180363b-36c4-416c-9624-d162aa131394">
<img width="1171" alt="image" src="https://github.com/coronasafe/care_fe/assets/46787056/28740d3c-dd5a-440b-ab9d-868657a55fa1">
<img width="1170" alt="image" src="https://github.com/coronasafe/care_fe/assets/46787056/16d3706f-e523-4f21-8652-c00822178d5c">
<img width="1169" alt="image" src="https://github.com/coronasafe/care_fe/assets/46787056/ed6fc66b-7505-4081-bd4f-140937007125">
<img width="1173" alt="image" src="https://github.com/coronasafe/care_fe/assets/46787056/5f9da169-6642-4bf5-97a6-2dc145f6176d">
<img width="1172" alt="image" src="https://github.com/coronasafe/care_fe/assets/46787056/b71da410-d331-4435-a634-c3e1783b4b7b">
<img width="1170" alt="image" src="https://github.com/coronasafe/care_fe/assets/46787056/30efb21e-58b0-4a06-9e66-a7e71b00c3b9">
<img width="1169" alt="image" src="https://github.com/coronasafe/care_fe/assets/46787056/fbbccd3c-d679-486c-be7c-cfff9be36edb">
<img width="1173" alt="image" src="https://github.com/coronasafe/care_fe/assets/46787056/93997a6b-92d9-4a47-b346-bc5da16ddde7">

very rough sketch: 

<img width="1103" alt="image" src="https://github.com/coronasafe/care_fe/assets/46787056/202254c1-ed65-4c23-ac52-f1735b028d1d">

cc: @aparnacoronasafe

**Comment by saloni0419** [comment]
@sainak @rithviknishad  can you please assign this issue to me

**Comment by sainak** [comment]
@saloni0419 wait for a thumbs up from @aparnacoronasafe

**Comment by saloni0419** [comment]
Sure

**Comment by ab1123** [comment]
I would like to work on this issue. I will focus on making the header a separate component. In order to make the header reusable I would like to use the concept of barrel export with the header component. If you kindly allow me to work on this issue @sainak @aparnacoronasafe @rithviknishad

**Comment by SinghYuvraj0506** [comment]
Hi, @sainak can I work on the issue, I would create a new general header component that would fit most of the types shown, for any extra case we can create a separate functional component  in the same component file

**Comment by ab1123** [comment]
@sainak @aparnacoronasafe any updates on this issue??

**Comment by kumar11jr** [comment]
Hii @sainak can i work on this issue.

**Comment by UdaySagar-Git** [comment]
I would like to work on this , can you please assign me

**Comment by aparnacoronasafe** [comment]
Please go ahead. 

However, the suggested design is missing some elements. like the "Live / Discharge" tabs. 

Please be mindful of retaining all the elements in the headers

**Comment by sainak** [comment]
@saloni0419 @ab1123 @SinghYuvraj0506 @kumar11jr @UdaySagar-Git 
Post your mockups and implementation plan before working on this issue

cc: @Ashesh3 @rithviknishad @nihal467

**Comment by Shahbaz898414** [comment]
**Hii @sainak can i work on this issue.**

**Comment by bishwas-10** [comment]
hi @sainak can i work on this issue? 
my implementation plan will be :

-i will make a reusable component named Header in a Common/components/ directory named Header.js file
-and another component named HeaderCountBlock in CAREUI/display/Count.js directory for no of counts of each sections like results sample like this
![image](https://github.com/coronasafe/care_fe/assets/120932783/42ceffd8-d418-42d6-8ca8-f7b08e3f6a8e)
-for the header section each component like searchbox , advanced filter , switch tabs will be sent as a props and will render 

and the final result may look like this 
![image](https://github.com/coronasafe/care_fe/assets/120932783/5f5170fe-49ff-447d-b11d-c979ace1e58b)
![image](https://github.com/coronasafe/care_fe/assets/120932783/329b302a-e90a-457e-b89f-e32708444431)
![image](https://github.com/coronasafe/care_fe/assets/120932783/2af93533-d296-4c9c-8d72-0ccd8668ec26)
  will be changed according to your suggestions!!

**Comment by yashdargude** [comment]
Hii @sainak , I would like to work on this , can you please assign me

**Comment by rithviknishad** [comment]
> hi @sainak can i work on this issue? my implementation plan will be :
> 
> -i will make a reusable component named Header in a Common/components/ directory named Header.js file -and another component named HeaderCountBlock in CAREUI/display/Count.js directory for no of counts of each sections like results sample like this ![image](https://private-user-images.githubusercontent.com/120932783/310904891-42ceffd8-d418-42d6-8ca8-f7b08e3f6a8e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk4MjU2NTksIm5iZiI6MTcwOTgyNTM1OSwicGF0aCI6Ii8xMjA5MzI3ODMvMzEwOTA0ODkxLTQyY2VmZmQ4LWQ0MTgtNDJkNi04Y2E4LWY3YjA4ZTNmNmE4ZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMwN1QxNTI5MTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iMzU2ZmY2Yjk3MzI2Mjc3MTdmYjBhNjRiYTE3MDE1ZmFiZWMwYWVlNjk0MWRiYmNkYWUyN2VlMjgzMDdkMmZmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.NNO1cllX3ZiXhnkT3izwtqBsosj5CB6mw3wkH9q9BvA) -for the header section each component like searchbox , advanced filter , switch tabs will be sent as a props and will render
> 
> and the final result may look like this ![image](https://private-user-images.githubusercontent.com/120932783/310905937-5f5170fe-49ff-447d-b11d-c979ace1e58b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk4MjU2NTksIm5iZiI6MTcwOTgyNTM1OSwicGF0aCI6Ii8xMjA5MzI3ODMvMzEwOTA1OTM3LTVmNTE3MGZlLTQ5ZmYtNDQ3ZC1iMTFkLWM5NzlhY2UxZTU4Yi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMwN1QxNTI5MTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iOWM3MzBiZGNiOTlkMWQ4MWVhZTg2MTAyNjAxNTdhYjUxY2VlMzE0ODJhZDIzNjllZTY2MWZmODJlMTU2MzMwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.c4gTSEoD3HW7BQK1beenCw6vsjsw8N3rZyE51ZLjFxo) ![image](https://private-user-images.githubusercontent.com/120932783/310906026-329b302a-e90a-457e-b89f-e32708444431.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk4MjU2NTksIm5iZiI6MTcwOTgyNTM1OSwicGF0aCI6Ii8xMjA5MzI3ODMvMzEwOTA2MDI2LTMyOWIzMDJhLWU5MGEtNDU3ZS1iODlmLWUzMjcwODQ0NDQzMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMwN1QxNTI5MTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05YWZkYjc5YmZmODkyNmUyMTc5MmVmMzQ5YWE4OTU0OWE1ZTA5Yjk3NzgyZDIxMTk3ZThjZWFkMDBhOTkwZjFhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.pMhjmrIONGwu_oZlVt2R74NQY3Rmj7CxGOz6WMDfsmg) ![image](https://private-user-images.githubusercontent.com/120932783/310906075-2af93533-d296-4c9c-8d72-0ccd8668ec26.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk4MjU2NTksIm5iZiI6MTcwOTgyNTM1OSwicGF0aCI6Ii8xMjA5MzI3ODMvMzEwOTA2MDc1LTJhZjkzNTMzLWQyOTYtNGM5Yy04ZDcyLTBjY2Q4NjY4ZWMyNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMwN1QxNTI5MTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00NzMwNTk0YmE4YzEwZTUyNzQwNTNhOTg1ZjFiY2JlNWFlZDU2ZmM0Yjg2Yzk1Njc2YWMzMTdiMDAwNGQ2YTU3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.BftvNq3wt0MJB9zsGxNkTls7XZO1hHqSIgXzmuFdWVA) will be changed according to your suggestions!!

@bishwas-10 this still does not look consistent at all across all pages. 

The objective is to have consistent placement of common elements like filter buttons should be present in the same location on every page. Same with Import/Export, Add (Patient | Asset | ...), etc.

**Comment by bishwas-10** [comment]
@rithviknishad ok i will work on that as i  mentioned above will be changed according to your suggestion.
And it will be easy for me to work on this if you guys assign me this issue

**Comment by rithviknishad** [comment]
> @rithviknishad ok i will work on that as i mentioned above will be changed according to your suggestion. And it will be easy for me to work on this if you guys assign me this issue

https://github.com/coronasafe/care_fe/issues/7316#issuecomment-1981128727

**Comment by SinghYuvraj0506** [comment]
@sainak @rithviknishad , i have built a mockup considering all the above conditions
 
<img width="831" alt="Screenshot 2024-03-08 at 1 11 10 AM" src="https://github.com/coronasafe/care_fe/assets/98007961/b53dc94e-49db-486a-8b06-068a576adfff">

From the above images of headers we currently have, I found some points :-
1. There are some fields which are common on every header like export/import , filters , a search field, section name, counts
2. Some CTA are present on some headers, these are different for different sections .
3. Some pages contain some switches for parsing different data.
4. Some pages contain more than one search fields.

The mockup solves all these problems,:
1. The fixed fields are kept constant at one corner.
2. These CTA can be added according to our needs and are optional, can handle more buttons too in a grid system
3. We can fix a switch on the topmost side of the header which is optional
4. I have provided a dropdown in the search field, where one can select the search prop to be searched and add more such search fields.

in technical aspect
Let's create a header component capable of all these features, it would have props that would control all these features like:

<img width="691" alt="image" src="https://github.com/coronasafe/care_fe/assets/98007961/54da6063-eee9-4342-a4dc-f5f0d2d01edc">


cc: @nihal467 @Ashesh3

**Comment by rithviknishad** [comment]
@SinghYuvraj0506 This looks great and example code seems to be very reusable and would lead to a consistent design pattern.

Just letting you know that, we generally refrain from "Icon Only" designs. It'd be great if labels are also present along with the icons so any users who may not be able to interpret icons would also find it easy to use.

**Comment by SinghYuvraj0506** [comment]
@rithviknishad  i would use labels instead of icons, or we could even use the tip to tell about the tool over hovering it.
Also we need a responsive mobile UI , I think it would be best if we could align all the features of top row of the design I provided into hamburger menu and rest could be adjusted

**Comment by rithviknishad** [comment]
But we already have a hamburger menu for mobile. Did you mean to have two hamburger menus?

**Comment by Shahbaz898414** [comment]
Hii @sainak , I would like to work on this , can you please assign me

**Comment by SinghYuvraj0506** [comment]
I have made all the possibilities of the header in the design 

<img width="830" alt="Screenshot 2024-03-15 at 5 00 08 PM" src="https://github.com/coronasafe/care_fe/assets/98007961/7bb72c6d-8308-4d88-ac29-83587329c962">


Hey @rithviknishad @sainak @nihal467  look into the design and if it is correct , then I would to like to work on the issue, could you please assign it to me

**Comment by vishwansh01** [comment]
@sainak  @rithviknishad  can you assign this issue to me ?

**Comment by Shahbaz898414** [comment]
Hii @sainak , I would like to work on this , can you please assign me

**Comment by ishaanxgupta** [comment]
Hi @sainak, I would like to work on this issue, please assign.
Thanks

**Comment by manmeetnagii** [comment]
@rithviknishad  I noticed the issue about enhancing all the headers across the pages. I’d like to take up this task.

In my recent PR #9392 , I worked on enhancing the header of the Asset page, which gave me a deeper understanding of the codebase and the header structure. For this task, I will:

- Create a reusable header component using ShadCN UI button and SearchByMultipleFields.

 I would like to work on this issue.

**Comment by manmeetnagii** [comment]
@Jacobjeevan can I create an issue for enhancing the Shifting Page Header Layout to align it with the Facilities and Patient header layouts. Also, I will ensure to use shadcn UI components.

**Comment by Jacobjeevan** [comment]
> [@Jacobjeevan](https://github.com/Jacobjeevan) can I create an issue for enhancing the Shifting Page Header Layout to align it with the Facilities and Patient header layouts. Also, I will ensure to use shadcn UI components.

No need to create a new issue; I can assign this one to you 👍

**Comment by manmeetnagii** [comment]
@Jacobjeevan sure, thank you

**Comment by manmeetnagii** [comment]
@Jacobjeevan please check this.What we can add in resources count 🤔

## Desktop View
![Image](https://github.com/user-attachments/assets/9e747b33-3f5f-4b1e-a2bd-e508f9e6bd16)
![Image](https://github.com/user-attachments/assets/7255faee-a5d1-489e-936b-88f02bdbe9de)
![Image](https://github.com/user-attachments/assets/f21508ce-8db5-4278-9135-8347f7120dea)
![Image](https://github.com/user-attachments/assets/8a8a641c-6a5f-4bab-86e8-65b5437a2cd1)
![Image](https://github.com/user-attachments/assets/c032d965-b1ca-4f86-b977-42e4b93bcdd9)
![Image](https://github.com/user-attachments/assets/bf5a4d92-00e2-4fcd-bc14-b65c5a114102)


## Mobile View
![Image](https://github.com/user-attachments/assets/43cb3661-e5b4-427a-b423-ea55fd921837)
![Image](https://github.com/user-attachments/assets/184682db-1519-4fb4-b9ed-a6ee5978b235)
![Image](https://github.com/user-attachments/assets/a869ea57-323a-4ff9-b7c2-18576a219f77)
![Image](https://github.com/user-attachments/assets/641bc637-112c-40a3-b6d3-494282702718)
![Image](https://github.com/user-attachments/assets/52f93635-46d7-48c5-8098-39138fbc19c7)
![Image](https://github.com/user-attachments/assets/8292c21b-e557-4da3-aa08-50c27c6b04dd)

**Comment by rithviknishad** [comment]
before you work on this, do checkout: 

- https://github.com/ohcnetwork/care_fe/pull/9262
- https://github.com/ohcnetwork/care_fe/pull/9400
- https://github.com/ohcnetwork/care_fe/pull/9439

things have changed quite a bit over there.

**Comment by manmeetnagii** [comment]
Do I have to make changes according to these PRs' codebase? or wait to merge these

**Comment by rithviknishad** [comment]
i'd recommend hold this for now, and take this up if necessary after #9262 is merged.

**Comment by jeevjacobgeorge** [comment]
let me help

**Comment by manmeetnagii** [comment]
> i'd recommend hold this for now, and take this up if necessary after [#9262](https://github.com/ohcnetwork/care_fe/pull/9262) is merged.

@rithviknishad Okay , I will wait for the merge. However, I have gone through these PRs and set up the form-field-v1 branch locally.

**Comment by nihal467** [comment]
most of the pages are designed already, closing this issue
